private/buf/buflsp/completion.go

```diff
@@ -274,26 +411,122 @@ func completionItemsForImport(ctx context.Context, file *file, declImport ast.De
+// topLevelKeywords returns keywords for the top-level.
+func topLevelKeywords() iter.Seq[keyword.Keyword] {
+	return func(yield func(keyword.Keyword) bool) {
+		_ = yield(keyword.Syntax) &&
+			yield(keyword.Edition) &&
+			yield(keyword.Import) &&
+			yield(keyword.Package) &&
+			yield(keyword.Message) &&
+			yield(keyword.Service) &&
+			yield(keyword.Option) &&
+			yield(keyword.Enum) &&
+			yield(keyword.Extend)
+	}
+}
+
+// messageLevelFieldKeywords returns keywords for messages.
+func messageLevelKeywords(isProto2 bool) iter.Seq[keyword.Keyword] {
+	return func(yield func(keyword.Keyword) bool) {
+		ok := yield(keyword.Message) &&
+			yield(keyword.Enum) &&
+			yield(keyword.Option) &&
+			yield(keyword.Extend) &&
+			yield(keyword.Oneof) &&
+			yield(keyword.Extensions) &&
+			yield(keyword.Reserved)
+		_ = ok && isProto2 &&
+			yield(keyword.Group)
+	}
+}
+
+// messageLevelFieldKeywords returns keywords for type modifiers.
+func messageLevelFieldKeywords() iter.Seq[keyword.Keyword] {
+	return func(yield func(keyword.Keyword) bool) {
+		_ = yield(keyword.Repeated) &&
+			yield(keyword.Optional) &&
+			yield(keyword.Required)
+	}
+}
+
+// predeclaredTypeKeywords returns keywords for all predeclared types.
+func predeclaredTypeKeywords() iter.Seq[keyword.Keyword] {
+	return func(yield func(keyword.Keyword) bool) {
+		_ = yield(keyword.Int32) &&
+			yield(keyword.Int64) &&
+			yield(keyword.UInt32) &&
+			yield(keyword.UInt64) &&
+			yield(keyword.SInt32) &&
+			yield(keyword.SInt64) &&
+			yield(keyword.Fixed32) &&
+			yield(keyword.Fixed64) &&
+			yield(keyword.SFixed32) &&
+			yield(keyword.SFixed64) &&
+			yield(keyword.Float) &&
+			yield(keyword.Double) &&
+			yield(keyword.Bool) &&
+			yield(keyword.String) &&
+			yield(keyword.Bytes)
+	}
+}
+
+// serviceLevelKeywords returns keywords for service.
+func serviceLevelKeywords() iter.Seq[keyword.Keyword] {
+	return func(yield func(keyword.Keyword) bool) {
+		_ = yield(keyword.RPC) &&
+			yield(keyword.Option)
+	}
+}
+
+// enumLevelKeywords returns keywords for enums.
+func enumLevelKeywords() iter.Seq[keyword.Keyword] {
+	return func(yield func(keyword.Keyword) bool) {
+		_ = yield(keyword.Option)
 	}
 }
 
 // keywordToCompletionItem converts a keyword to a completion item.
-func keywordToCompletionItem(kw keyword.Keyword) protocol.CompletionItem {
-	return protocol.CompletionItem{
-		Label: kw.String(),
-		Kind:  protocol.CompletionItemKindKeyword,
+func keywordToCompletionItem(
+	keywords iter.Seq[keyword.Keyword],
+	kind protocol.CompletionItemKind,
+	position protocol.Position,
+	prefix string,
+	suffix string,
+) iter.Seq[protocol.CompletionItem] {
+	return func(yield func(protocol.CompletionItem) bool) {
+		for keyword := range keywords {
+			suggest := keyword.String()
+			if !strings.HasPrefix(suggest, prefix) || !strings.HasSuffix(suggest, suffix) {
+				continue
+			}
+			if !yield(protocol.CompletionItem{
+				Label: suggest,
+				Kind:  kind,
```

Comment:
nit: I think it would be nice to add a `Detail` or `Documentation` field to these CompletionItems if we have it, that would match info shown in hover? (I'm not sure which field is correct and/or better.) Feel free to file this as a follow-up.

Once we get into custom types as well, I think we'll want to populate the `Deprecated` field if we know it.

<img width="218" height="56" alt="Image" src="https://github.com/user-attachments/assets/8f0ba281-6b94-4c87-b4b2-fb54e0b6c3c9" />

<img width="416" height="88" alt="Image" src="https://github.com/user-attachments/assets/849036d5-8ea1-4241-8bc2-2dda5a057b38" />